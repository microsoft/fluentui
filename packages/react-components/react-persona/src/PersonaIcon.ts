export * from './components/PersonaIcon/index';
