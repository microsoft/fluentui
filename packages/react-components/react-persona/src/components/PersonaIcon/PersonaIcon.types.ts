import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type PersonaIconSlots = {
  root: Slot<'div'>;
};

/**
 * PersonaIcon Props
 */
export type PersonaIconProps = ComponentProps<PersonaIconSlots> & {};

/**
 * State used in rendering PersonaIcon
 */
export type PersonaIconState = ComponentState<PersonaIconSlots>;
// TODO: Remove semicolon from previous line, uncomment next line, and provide union of props to pick from PersonaIconProps.
// & Required<Pick<PersonaIconProps, 'propName'>>
