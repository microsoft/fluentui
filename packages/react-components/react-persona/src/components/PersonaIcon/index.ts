export * from './PersonaIcon';
export * from './PersonaIcon.types';
export * from './renderPersonaIcon';
export * from './usePersonaIcon';
export * from './usePersonaIconStyles';
