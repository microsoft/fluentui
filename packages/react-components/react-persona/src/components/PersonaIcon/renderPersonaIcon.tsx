import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { PersonaIconState, PersonaIconSlots } from './PersonaIcon.types';

/**
 * Render the final JSX of PersonaIcon
 */
export const renderPersonaIcon_unstable = (state: PersonaIconState) => {
  const { slots, slotProps } = getSlots<PersonaIconSlots>(state);

  // TODO Add additional slots in the appropriate place
  return <slots.root {...slotProps.root} />;
};
