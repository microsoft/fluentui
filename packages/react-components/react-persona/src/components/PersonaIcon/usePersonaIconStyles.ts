import { makeStyles, mergeClasses } from '@griffel/react';
import type { PersonaIconSlots, PersonaIconState } from './PersonaIcon.types';
import type { SlotClassNames } from '@fluentui/react-utilities';

export const personaIconClassNames: SlotClassNames<PersonaIconSlots> = {
  root: 'fui-PersonaIcon',
  // TODO: add class names for all slots on PersonaIconSlots.
  // Should be of the form `<slotName>: 'fui-PersonaIcon__<slotName>`
};

/**
 * Styles for the root slot
 */
const useStyles = makeStyles({
  root: {
    // TODO Add default styles for the root element
  },

  // TODO add additional classes for different states and/or slots
});

/**
 * Apply styling to the PersonaIcon slots based on the state
 */
export const usePersonaIconStyles_unstable = (state: PersonaIconState): PersonaIconState => {
  const styles = useStyles();
  state.root.className = mergeClasses(personaIconClassNames.root, styles.root, state.root.className);

  // TODO Add class names to slots, for example:
  // state.mySlot.className = mergeClasses(styles.mySlot, state.mySlot.className);

  return state;
};
