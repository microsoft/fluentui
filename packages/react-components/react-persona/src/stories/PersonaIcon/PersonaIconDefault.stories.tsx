import * as React from 'react';
import { PersonaIcon, PersonaIconProps } from '@fluentui/react-persona';

export const Default = (props: Partial<PersonaIconProps>) => <PersonaIcon {...props} />;
