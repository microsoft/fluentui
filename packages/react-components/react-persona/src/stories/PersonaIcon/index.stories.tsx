import { PersonaIcon } from '@fluentui/react-persona';

import descriptionMd from './PersonaIconDescription.md';
import bestPracticesMd from './PersonaIconBestPractices.md';

export { Default } from './PersonaIconDefault.stories';

export default {
  title: 'Preview Components/PersonaIcon',
  component: PersonaIcon,
  parameters: {
    docs: {
      description: {
        component: [descriptionMd, bestPracticesMd].join('\n'),
      },
    },
  },
};
