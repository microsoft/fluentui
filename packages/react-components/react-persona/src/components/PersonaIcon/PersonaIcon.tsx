import * as React from 'react';
import { usePersonaIcon_unstable } from './usePersonaIcon';
import { renderPersonaIcon_unstable } from './renderPersonaIcon';
import { usePersonaIconStyles_unstable } from './usePersonaIconStyles';
import type { PersonaIconProps } from './PersonaIcon.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * PersonaIcon component - TODO: add more docs
 */
export const PersonaIcon: ForwardRefComponent<PersonaIconProps> = React.forwardRef((props, ref) => {
  const state = usePersonaIcon_unstable(props, ref);

  usePersonaIconStyles_unstable(state);
  return renderPersonaIcon_unstable(state);
});

PersonaIcon.displayName = 'PersonaIcon';
