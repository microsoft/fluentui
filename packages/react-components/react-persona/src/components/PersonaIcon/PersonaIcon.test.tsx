import * as React from 'react';
import { render } from '@testing-library/react';
import { PersonaIcon } from './PersonaIcon';
import { isConformant } from '../../common/isConformant';

describe('PersonaIcon', () => {
  isConformant({
    Component: PersonaIcon,
    displayName: 'PersonaIcon',
  });

  // TODO add more tests here, and create visual regression tests in /apps/vr-tests

  it('renders a default state', () => {
    const result = render(<PersonaIcon>Default PersonaIcon</PersonaIcon>);
    expect(result.container).toMatchSnapshot();
  });
});
