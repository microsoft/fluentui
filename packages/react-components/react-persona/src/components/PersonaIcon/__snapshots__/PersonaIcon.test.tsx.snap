// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonaIcon renders a default state 1`] = `
<div>
  <div
    class="fui-PersonaIcon"
  >
    Default PersonaIcon
  </div>
</div>
`;
